site-builder/src/site/resource.rs

```diff
@@ -240,6 +240,31 @@ pub(crate) struct ResourceSet {
 }
 
 impl ResourceSet {
+    /// Calculates unchanged resources from the new resources by checking
+    /// the `update_operations` for created or deleted resources.
+    ///
+    /// This approach leverages `BTreeSet` for lookups and filtering.
+    /// Calculates unchanged resources based on the new resources (passed in `self`)
+    /// and the update operations (to find which ones were not created or deleted).
+    pub fn calculate_unchanged_resources<'a>(
```

Comment:
Yes, I think this could be a good option if it’s better to maintain consistency in the way operations are used. Should I try to implement this approach and make all the necessary changes to include the new resource operation?

Reply:
that would be great! thank you @chariskms